sdk/messaging/azservicebus/sender.go

```diff
@@ -39,33 +40,65 @@ type MessageBatchOptions struct {
 // messages. Sending a batch of messages is more efficient than sending the
 // messages one at a time.
 func (s *Sender) NewMessageBatch(ctx context.Context, options *MessageBatchOptions) (*MessageBatch, error) {
-	sender, _, _, _, err := s.links.Get(ctx)
+	var lastRevision uint64
+	var batch *MessageBatch
 
-	if err != nil {
-		return nil, err
-	}
+	err := internal.Retry(ctx, "send", func(ctx context.Context, args *internal.RetryFnArgs) error {
+		if args.LastErr != nil {
+			if err := s.links.RecoverIfNeeded(ctx, lastRevision, args.LastErr); err != nil {
+				return err
+			}
+		}
+
+		sender, _, _, lr, err := s.links.Get(ctx)
```

Comment:
I assume this is the thing we're retrying as it does network I/O

Reply:
Yes, we could minimize the block but I mostly just like to make it simpler to read through.